### PR TITLE
Improve backends documentation

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Inspired by #14543.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: tutorials/introductory/usage.py [hunk 1]
```diff
@@ -473,10 +493,25 @@ def my_plotter(ax, data1, data2, param_dict):
 # :mod:`matplotlib` first tries to import it, if the import fails, it tries to
 # import `PySide`.
 #
+# Using non-builtin backends
+# --------------------------
+# 3rd parties can provide additional backends. Some of them (e.g. jupyter and
+# pycharm have their own way of activation). However, there is also a generic
```
Comment (anntzer) [inline tutorials/introductory/usage.py]: move parenthesis earlier

File: tutorials/introductory/usage.py [hunk 2]
```diff
@@ -473,10 +493,25 @@ def my_plotter(ax, data1, data2, param_dict):
...
+# way to activate non-builtin backends. If you have a module ``my_backend``
```
Comment (anntzer) [inline tutorials/introductory/usage.py]: dot -> colon

Comment (anntzer) [inline tutorials/introductory/usage.py]: "More generally, any importable backend can be selected by using any of the methods above with `module://name.of.the.backend` as the backend name, e.g. `matplotlib.use("module://name.of.the.backend")`."

File: tutorials/introductory/usage.py [hunk 3]
```diff
@@ -340,9 +356,9 @@ def my_plotter(ax, data1, data2, param_dict):
 #    use a different backend.  Therefore, you should avoid explicitly calling
 #    `~matplotlib.use` unless absolutely necessary.
 #
-# .. note::
-#    Backend name specifications are not case-sensitive; e.g., 'GTK3Agg'
-#    and 'gtk3agg' are equivalent.
+#
+# The builtlin backends
```
Comment (anntzer) [inline tutorials/introductory/usage.py]: typo

File: tutorials/introductory/usage.py [hunk 4]
```diff
@@ -290,44 +290,60 @@ def my_plotter(ax, data1, data2, param_dict):
...
-# #. Setting the :envvar:`MPLBACKEND` environment variable, either for your
-#    current shell or for a single script.  On Unix::
+#    On Unix::
 #
-#         > export MPLBACKEND=module://my_backend
+#         > export MPLBACKEND=qt5agg
 #         > python simple_plot.py
 #
-#         > MPLBACKEND="module://my_backend" python simple_plot.py
+#         > MPLBACKEND="qt5agg" python simple_plot.py
```
Comment (anntzer) [inline tutorials/introductory/usage.py]: quotes not needed (either there or when using module://) (note that on Windows you don't use quotes either, and if anything one should prefer single to double quotes in the shell anyways...)

File: tutorials/introductory/usage.py [hunk 5]
```diff
@@ -340,9 +356,9 @@ def my_plotter(ax, data1, data2, param_dict):
...
 #    `~matplotlib.use` unless absolutely necessary.
 #
-# .. note::
-#    Backend name specifications are not case-sensitive; e.g., 'GTK3Agg'
-#    and 'gtk3agg' are equivalent.
+#
+# The builtin backends
+# --------------------
 #
 # With a typical installation of matplotlib, such as from a
```
Comment (anntzer) [inline tutorials/introductory/usage.py]: this paragraph isn't really true anymore now that we have autodetection

File: tutorials/introductory/usage.py [hunk 6]
```diff
@@ -424,6 +440,10 @@ def my_plotter(ax, data1, data2, param_dict):
 #           This backend can be activated in IPython with ``%matplotlib wx``.
 # ========= ================================================================
 #
+# .. note::
+#    Backend name specifications are case-insensitive; e.g., 'Qt5Agg'
```
Comment (anntzer) [inline tutorials/introductory/usage.py]: *builtin* backend names

File: tutorials/introductory/usage.py [hunk 7]
```diff
@@ -531,9 +560,9 @@ def my_plotter(ax, data1, data2, param_dict):
 #     plt.xlabel("index")
 #
 # and you will see the plot being updated after each line.  Since version 1.5,
-# modifying the plot by other means *should* also automatically
-# update the display on most backends. Get a reference to the :class:`~matplotlib.axes.Axes` instance,
-# and call a method of that instance::
+# modifying the plot by other means *should* also automatically update the
```
Comment (anntzer) [inline tutorials/introductory/usage.py]: I would reword this and the paragraph above to not mention mpl<1.5 (well that can be left to a note if you want to keep it, e.g. "note that if you are running mpl<1.5, etc.")

File: tutorials/introductory/usage.py (hunk 5 above)
Comment (timhoffm) [reply tutorials/introductory/usage.py]: I don't feel competent enough describing what is currently happening with autodetection. Please push a suggestion or open a ticket to mark this as outdated.
